python/recipe.hpp

```diff
@@ -64,7 +64,12 @@ class py_recipe_trampoline: public py_recipe {
     }
 
     arb::cell_kind cell_kind(arb::cell_gid_type gid) const override {
-        PYBIND11_OVERRIDE_PURE(arb::cell_kind, py_recipe, cell_kind, gid);
+        try {
+            PYBIND11_OVERRIDE_PURE(arb::cell_kind, py_recipe, cell_kind, gid);
+        }
+        catch (const pybind11::cast_error& e) {
+            throw pybind11::type_error{"Couldn't convert return value of recipe.cell_kind(gid) to a cell_kind. Please check your recipe."};
```

Comment:
Is `cell_kind` a fully qualified unambiguous name of the expected return type? Maybe `to an arbor.cell_kind enum` or similar?

Reply:
Fully qualified within Arbor, yes. It should be enough of a hint, no?